### PR TITLE
Web mobile fix & initial settings page

### PR DESCRIPTION
**Pull request by madagas-karrh**
# Summary of changes
I made arclightning_web files look way better on mobile, as I'm assuming most users will access through mobile phone.
Also added initial settings page! Just sending POST and GET to arbitrary values right now until endpoints set up.
For issue #48 :
- [x] name: user's name. cap it at 200 bytes
- [x]  profile text color: color picker, used to display the name
- [x] profile background color: color picker, used to display the name
- [x] preview of name with text color and background color
- [x] volume slider. corresponds to a floating point value from 0.0 to 1.0
- [x] submitting the form to hostname/api/v1/edit_settings with json body {name, profile_fgcolor, profile_bgcolor, volume}

For issue #55:
- [x] on load, the settings page should hit the check_settings endpoint
- [ ] fill itself in using that data

For issue #58:
- [x] Registration and log in should link to each other

For issue #59:
- [x] settings page needs a button to log out

# How was this tested?
Tested on multiple screen sizes, not very good looking on big sizes, but mobile sizes look good 

Oh also covers issue #58 (links registration and login together through nav bar) 
Oh and issue #59, nav bar has log out on settings page

### CODE REVIEW

**Comment by madagas-karrh** [comment]
I dont know if this is what you were expecting for the settings page, so please tell me if something is wrong (I will fix the way stuff looks on large screens tomorrow)

**Comment by madagas-karrh** [comment]
okay im sorry i am the worst I did not realize these were all separate issues, I should have looked ahead. But whatever now its a jumbled mess, I hope you enjoy it

File: arclightning_web/login.html [hunk 1]
```diff
@@ -15,22 +20,50 @@
...
 </body>
+<script>
+    new Vue({
+        el: '#login',
+        props: ['username', 'pw',],
+        methods: {
+            login: function () {
+                    axios.post('http://hostname/api/v1/signin', JSON.stringify({ username: this.username, password: this.password }))
+			//should this redirect to settings page probably?
+                    alert("You're logged in!")
```
Comment (starfys) [inline arclightning_web/login.html]: go ahead and change this alert to a redirect to settings

File: arclightning_web/settings.html [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+               <br> <input type="text" maxlength="200" id="name" style=" top: 10%; text-align:center;width: 100%; border:none; border-bottom: 2px solid black;" v-model="name" v-bind:style="{color: profile_fgcolor, 'background-color': profile_bgcolor}"required /><br>
+
+		<h1 class="formattedText centerItem" >Name color:</h1>
+                <input class="centerItem" type="color" style="height: 50px; width:80%;" id="profile_fgcolor"  value="#333999" v-model="profile_fgcolor"/>
+
+		<h1 class="formattedText" style="display:inline">Background color:</h1>
+                <input class="centerItem" type="color" style="height: 50px; width: 80%;" id="profile_bgcolor" value="#219444" v-model="profile_bgcolor" />
+
+                <p class="formattedText">Volume: </p>
+		<input class="centerItem" type="range" id="volume" style="width: 80%;" min="0.0" max="1.0" step = "0.05" v-model="volume"/>
```
Comment (starfys) [inline arclightning_web/settings.html]: can we have this be one of those slider bars that fill up vs a dot on a line?

like this 
▮▮▮▮▮▮▯▯▯▯

Comment (madagas-karrh) [reply arclightning_web/settings.html]: So, I looked into this and I thiiiink we'd have to use an external library to make it available to all browsers. Do you want me to send you some of the libraries or leave it as is?

File: arclightning_web/login.html (hunk 1 above)
Comment (madagas-karrh) [reply arclightning_web/login.html]: fixed this!

**Pull request merged**